client/index.js

```diff
@@ -150,24 +151,51 @@ addFilter(
 			},
 			capability: 'manage_woocommerce',
 		} );
-		pages.push( {
-			container: DisputeDetailsPage,
-			path: '/payments/disputes/details',
-			wpOpenMenu: menuID,
-			breadcrumbs: [
-				rootLink,
-				[
-					'/payments/disputes',
-					__( 'Disputes', 'woocommerce-payments' ),
-				],
-				__( 'Dispute details', 'woocommerce-payments' ),
-			],
-			navArgs: {
-				id: 'wc-payments-disputes-details',
-				parentPath: '/payments/disputes',
-			},
-			capability: 'manage_woocommerce',
-		} );
+
+		// If disputes on transaction page feature is enabled, set up a soft
+		// redirect component; otherwise register the (legacy) dispute details page.
+		const isDisputeOnTransactionPageEnabled =
+			window.wcpaySettings.featureFlags.isDisputeOnTransactionPageEnabled;
+		pages.push(
+			isDisputeOnTransactionPageEnabled
+				? {
+						container: RedirectToTransactionDetails,
+						path: '/payments/disputes/details',
+						wpOpenMenu: menuID,
+						breadcrumbs: [
+							rootLink,
+							[
+								'/payments/disputes',
+								__( 'Disputes', 'woocommerce-payments' ),
+							],
+							__( 'Dispute details', 'woocommerce-payments' ),
+						],
+						navArgs: {
+							id: 'wc-payments-disputes-details-legacy-redirect',
+							parentPath: '/payments/disputes',
+						},
+						capability: 'manage_woocommerce',
+				  }
+				: {
+						container: DisputeDetailsPage,
+						path: '/payments/disputes/details',
+						wpOpenMenu: menuID,
+						breadcrumbs: [
+							rootLink,
+							[
+								'/payments/disputes',
+								__( 'Disputes', 'woocommerce-payments' ),
+							],
+							__( 'Dispute details', 'woocommerce-payments' ),
+						],
+						navArgs: {
+							id: 'wc-payments-disputes-details',
+							parentPath: '/payments/disputes',
+						},
+						capability: 'manage_woocommerce',
+				  }
+		);
```

Comment:
I was expecting only the `container` that needs to conditionally change between `RedirectToTransactionDetails` and `DisputeDetailsPage` based on whether `isDisputeOnTransactionPageEnabled` is on or not.

Why `navArgs.id` needs to be different?

Also, not sure what the correlation between this page registration in Javascript and [in PHP](https://github.com/Automattic/woocommerce-payments/blob/2eb4de5d4bca3573339e15197987a3943ffeddf4/includes/admin/class-wc-payments-admin.php#L468-L475).

Per the [docs](https://developer.woocommerce.com/extension-developer-guide/working-with-woocommerce-admin-pages/#registering-a-react-powered-page), react powered page needs to be registered both in Javascript and PHP.